lib/class-wp-theme-json-resolver.php

```diff
@@ -258,27 +258,39 @@ private static function get_core_origin() {
 	}
 
 	/**
-	 * Returns the theme's origin config.
+	 * Returns the theme's data.
 	 *
-	 * It uses the theme support data if
-	 * the theme hasn't declared any via theme.json.
+	 * Data from theme.json can be augmented via the
+	 * $theme_support_data variable. This is useful, for example,
+	 * to backfill the gaps in theme.json that a theme has declared
+	 * via add_theme_supports.
+	 *
+	 * Note that if the same data is present in theme.json
+	 * and in $theme_support_data, the theme.json's is not overwritten.
 	 *
 	 * @param array $theme_support_data Theme support data in theme.json format.
 	 *
 	 * @return WP_Theme_JSON Entity that holds theme data.
 	 */
-	private function get_theme_origin( $theme_support_data = array() ) {
-		$theme_json_data = self::get_from_file( locate_template( 'experimental-theme.json' ) );
-		self::translate_presets( $theme_json_data, wp_get_theme()->get( 'TextDomain' ) );
+	public function get_theme_data( $theme_support_data = array() ) {
+		if ( null === $this->theme ) {
+			$theme_json_data = self::get_from_file( locate_template( 'experimental-theme.json' ) );
+			self::translate_presets( $theme_json_data, wp_get_theme()->get( 'TextDomain' ) );
+			$this->theme = new WP_Theme_JSON( $theme_json_data );
+		}
```

Comment:
Regarding follow up improvements that were mentioned:  When we instantiate the class, it seems `$this->theme` is always `null`.  So if we are calling a `new WP_Theme_JSON_Resolver()` in one place for page templates, and another place for template parts (or anything else).. does that mean we are getting and translating the file separately every time?  Should `theme` itself be static?  Or maybe I am missing something already in place that prevents this from happening more than necessary?

Reply:
Yes! I pushed https://github.com/WordPress/gutenberg/pull/28700/commits/6ed01a9a91b581a58a10d894f9627a39c0c1bfb5 to fix it (I forgot about this, tbh).